File: DSAlgorithms/Leetcode/ts/src/arrays_hash/2215-find_dif_of_2_arr.ts
const findDifference  = (nums1: number[], nums2: number[]): number[][] =>{
    const set1 = new Set<number>();
    const set2 = new Set<number>();
    const ans1 = new Set<number>();
    const ans2 = new Set<number>();
    const res1: number[] = [];
    const res2: number[] = [];
    const result: number [][] = [];
    for(let i = 0; i < nums1.length; i++)
        set1.add(nums1[i]);
    for(let i = 0; i < nums2.length; i++)
        set2.add(nums2[i]);
    for(let i = 0; i < nums1.length; i++)
        if(!set2.has(nums1[i]))
            ans1.add(nums1[i]);
    for(let i = 0; i < nums2.length; i++)
        if(!set1.has(nums2[i]))
            ans2.add(nums2[i]);
    for(let item of ans1)
        res1.push(item);
    for(let item of ans2)
        res2.push(item);
    result.push(res1);
    result.push(res2);
    return result;
};

File: DSAlgorithms/Leetcode/c++/arrays_hash/2215-find_dif_of_2_arr.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        std::unordered_set<int> set1;
        std::unordered_set<int> set2;
        std::unordered_set<int> ans1;
        std::unordered_set<int> ans2;
        std::vector<int> res1;
        std::vector<int> res2;
        for(auto val : nums1)
            set1.insert(val);
        for(auto val: nums2)
            set2.insert(val);
        for(auto val: nums1)
        {
            std::unordered_set<int>::iterator it = set2.find(val);
            if(it == set2.end())
                ans1.insert(val);
        }
        for(auto val: nums2)
        {
            std::unordered_set<int>::iterator it = set1.find(val);
            if(it == set1.end())
                ans2.insert(val);
        }
        for(auto val: ans1)
           res1.push_back(val);
        for(auto val: ans2)
           res2.push_back(val);

        std::vector<std::vector<int>> res;
        res.push_back(res1);
        res.push_back(res2);

        return res;
    }
};